src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -175,21 +175,29 @@ public void visitToken(DetailAST ast) {
 
         if (rcurly != null) {
             final String violation;
-            if (shouldStartLine) {
-                final String targetSourceLine = getLines()[rcurly.getLineNo() - 1];
-                violation = validate(details, option, true, targetSourceLine);
-            }
-            else {
-                violation = validate(details, option, false, "");
-            }
-
+            violation = validate(details, option,
+                    shouldStartLine, getTargetSourceLine(rcurly.getLineNo() - 1));
             if (!violation.isEmpty()) {
                 log(rcurly, violation, "}", rcurly.getColumnNo() + 1);
             }
         }
     }
 
     /**
+     * Fetches line based on passed line no.
+     * @param lineNo for fetching lines.
+     * @return lines associated with line no.
+     */
+    private String getTargetSourceLine(int lineNo) {
```

Comment:
@romani Please confirm this for us.
Are you ok with code as it is changed now, or should we remove static from methods so `targetSourceLine` can be moved to where it is used, or do you think `targetSourceLine` should be added to `details` class.